### ISSUE TITLE
unexpected error received while funding wallet send coins

### ISSUE THREAD

**Issue by GlenDC** [rivine/rivine#313]
This is an issue that relates to the` feat-break-txns` branch, not (yet) to master.

While I have enough coins, the wallet seems to return an error, disagreeing with that.

```
$ rivinec wallet
Wallet status:
Encrypted, Unlocked
Confirmed Balance:   1222 ROC
Locked Balance:      50 ROC
Unconfirmed Delta:   + 0 ROC
BlockStakes:         1000000 BS

$ rivinec wallet send coins 010a6faecb46583cac9f2ca7d94a365b2686b6970669dd185a7bb8d1c104f3495c1b2324a958ea 1200
Could not send coins: error after call to /wallet/coins: wallet has coins spent in incomplete transactions - not enough remaining coins
```

**Comment by GlenDC** [comment]
Another error:

```
 $ rivinec wallet send coins 0161fbcf58efaeba8813150e88fc33405b3a77d51277a2cdf3f4d2ab770de287c7af9d456c4e68 47
Could not send coins: error after call to /wallet/coins: consensus conflict: unexpected unlock fulfillment
$ rivinec wallet
Wallet status:
Encrypted, Unlocked
Confirmed Balance:   48 ROC
Locked Balance:      0 ROC
Unconfirmed Delta:   + 0 ROC
BlockStakes:         0 BS
```

**Comment by GlenDC** [comment]
All these errors are possible since the implementation for #311 in 61c55cd.

**Comment by GlenDC** [comment]
> consensus conflict: unexpected unlock fulfillment

This error is simple to solve. As we now can spend unlock conditions of different (but known) types, we should check the type and create a different fulfillment depending upon the unlock condition we wish to spend.

**Comment by GlenDC** [comment]
> `error after call to /wallet/coins: wallet has coins spent in incomplete transactions - not enough remaining coins`

Seems like this is a result of the fact that our coin-selection algorithm is less than ideal (see #310), making it use the biggest available unspent output I have for the payment of the transaction fee while creating a block. By doing so, it keeps that output busy forever, since our logic doesn't use it for 40 blocks after spending it. And as this is the only fund available, big enough, to satisfy our request, we get the incomplete error. Expected after all.

**Comment by GlenDC** [comment]
closed by 45f4a0e
